rskj-core/src/main/java/co/rsk/peg/RepositoryBtcBlockStoreWithCache.java

```diff
@@ -132,7 +197,48 @@ public StoredBlock getStoredBlockAtMainChainHeight(int height) throws BlockStore
         StoredBlock chainHead = getChainHead();
         int depth = chainHead.getHeight() - height;
         logger.trace("Getting btc block at depth: {}", depth);
-        return getStoredBlockAtMainChainDepth(depth);
+
+        if (depth < 0) {
```

Comment:
If this case is possible, what happens now? It is not throwing an exception but returning null. Does the caller handle it the same way?

Reply:
I'd probably do it slightly different. Just add before this a `if (!activations.isActive(...)) { return getStoredBlockAtMainChainDepth(depth))` and then check for `depth < 0` and all the following code with no more activations guards.